features/support/form_helpers.rb

```diff
@@ -361,18 +361,23 @@ def open_categories(categories_headings)
   end
 
   def find_and_click_submit_button
-    submit_button = find_elements_by_xpath("//input[@class='button-save']")[0].value
-    if submit_button == 'Save and continue'
-      click_on 'Save and continue', wait: false
-      false
-    else
-      click_on submit_button, wait: false
+    # Get the submit buttons on the page, and categorise by whether we expect to redirect to the summary or not
+    save_and_return_button = find_elements_by_xpath("//button[contains(normalize-space(text()), 'Save and return')] | //button[contains(normalize-space(text()), 'Save and return to service summary')]")[0]
+    save_and_continue_button = find_elements_by_xpath("//button[contains(normalize-space(text()), 'Save and continue')]")[0]
```

Comment:
I think this can be simplified using the `find_button` method

```suggestion
    save_and_continue_button = find_button("Save and continue")
```

Reply:
This fails if it can't find the button, right? This helper function needs to look for both types of button in order to figure out whether it's the end of the section or not, so including this line fails on the company registration details for me.

Reply:
Ah yes, good point.

This will do it `first(:button, "Save and continue", minimum: 0)`, although I'm not sure whether it is more readable though.

Reply:
We use `find_elements_by_xpath` all over the place, mostly because it has the `synchronize_with_unload_wait` fix in it. I'm happy to keep it?